pyControl4/lock.py

```diff
@@ -0,0 +1,86 @@
+class C4Lock:
+    def __init__(self, C4Director, item_id):
+        """Creates a Control4 Lock object.
+
+        Parameters:
+            `C4Director` - A `pyControl4Ex.director.C4Director` object that
+            corresponds to the Control4 Director that the Lock is connected to.
+
+            `item_id` - The Control4 item ID of the Lock.
+        """
+        self.director = C4Director
+        self.item_id = item_id
+
+    async def is_locked(self):
+        """Returns True if it is currently Locked or False if it is Unlocked
+
+        Notes:
+            Relay State 0 == Locked
+            Relay State 1 == Unlocked
+        """
+        return not bool(
+            await self.director.getItemVariableValue(self.item_id, "RelayState")
+        )
+
+    async def state_verified(self):
+        """Returns True if Lock Relay is functional
+
+        Notes:
+            I think this is just used to verify that the relay is functional,
+            not 100% sure though.
```

Comment:
Do you have any other information on this or any way to test what conditions can cause this variable to change? (ie. can you force the relay to not be functional somehow so you can see what Control4 says?)

Reply:
I tried putting the lock in the middle of unlocked and locked, but that didn't change the State Verified value. I can't currently test that as I would probably have to take apart the lock itself to change it.

Reply:
Ok, I think we'll just leave this as is then.